hunk 1: allow finding X libs

hunk 2: find_library_file (which comes from distutils via setuptools)
has a hardcoded 'shared_lib_extension = ".so"' and only searches for the
full filename, so it misses OpenBSD-style libfoo.so.X.Y unless there's
also a static library with a matching name (but that was removed in an
update of graphics/tiff). Patch to workaround.

Index: setup.py
--- setup.py.orig
+++ setup.py
@@ -673,6 +673,12 @@ class pil_build_ext(build_ext):
                     ),
                 )
 
+        elif sys.platform.startswith("openbsd"):
+            _add_directory(library_dirs, "/usr/X11R6/lib")
+            _add_directory(library_dirs, "/usr/local/lib")
+            _add_directory(include_dirs, "/usr/X11R6/include")
+            _add_directory(include_dirs, "/usr/local/include")
+
         elif sys.platform.startswith("netbsd"):
             _add_directory(library_dirs, "/usr/pkg/lib")
             _add_directory(include_dirs, "/usr/pkg/include")
@@ -793,6 +799,8 @@ class pil_build_ext(build_ext):
                 ):
                     feature.set("tiff", "libtiff")
                 elif _find_library_file(self, "tiff"):
+                    feature.set("tiff", "tiff")
+                elif sys.platform.startswith("openbsd"):
                     feature.set("tiff", "tiff")
 
         if feature.want("freetype"):
